### ISSUE TITLE
Resource Local to Scene Flag causes Post Import script to convert local Resource from External to Built-in

### ISSUE THREAD

**Issue by christopwner** [godotengine/godot#97127]
### Tested versions

v4.3.stable.official [77dcf97d8]

### System information

Godot v4.3.stable - Ubuntu 24.04.1 LTS 24.04 - X11 - Vulkan (Forward+) - integrated AMD Radeon Graphics (RADV RENOIR) - AMD Ryzen 7 7730U with Radeon Graphics (16 Threads)

### Issue description

Setting the flag, local to scene, on an external resource will cause a post import script (EditorScenePostImport) that loads that resource to convert it from external to built-in for that scene. Expected resource to remain external until runtime which then a duplicate of the resource would be made for it's scene.

Expected:
![expected](https://github.com/user-attachments/assets/6f3ec562-2343-4fb4-ad19-b54c54e93b75)

Actual:
![actual](https://github.com/user-attachments/assets/a1626d40-3b5b-48dc-a763-bb2324b0199f)

### Steps to reproduce

1. Unzip and open project
4. View `cube/cube.tscn` to confirm shared `Integrity` resource on meshes `Cube_Healthy` and `Cube_Damaged` loaded as external
5. Open `cube/cube_integrity.tres` and set `Local to Scene` to true
6. Reimport by opening `cube/cube.glb` and clicking button at bottom of import dialog
7. Review `cube/cube.tscn` to view shared `Integrity` resource on meshes is now a built-in rather then reference the external.

### Minimal reproduction project (MRP)

[load-external-resource-as-local.zip](https://github.com/user-attachments/files/17034633/load-external-resource-as-local.zip)

**Comment by AThousandShips** [comment]
Not sure if this is a feature or a bug, it would make sense for this change to be respected, but not familiar with the internals here

CC @godotengine/import

**Comment by christopwner** [comment]
yeah, my assumption is that this is an unintended side-effect of the `@tool` annotation when loading the resource in post import script but would love to hear more from the team

**Comment by jitspoe** [comment]
This isn't just an issue with tool scripts.  The editor does this as well when you make instances of local to scene resources, as I've reported here: https://github.com/godotengine/godot/issues/69424
